pyscriptjs/src/components/base.ts

```diff
@@ -123,20 +130,20 @@ export class BaseEvalElement extends HTMLElement {
 
             this._register_esm(runtime);
             if (is_async) {
-                <string>await runtime.runPythonAsync(
+                <string>await runtime.runCodeAsync(
```

Comment:
I'm not sure we should differentiate `runAsync` from `run` and least not for the interface or a runtime API and not how it's intended here.

My understanding is that from the `pyodide` point of view we don't need to (@rth @hoodmane, please keep me honest here) and from a Facade API point of view (which is the exercise here) I'd rather have one `run` method that accepts a generic `options` mapping that can add flexibility to add specifics to each runtime so that underneath you can decide whether to run pyodide `runPython` or `runPythonAsync`. 

That said, I'd still add a `runAsync` in the future but that would probably mean that we are running the runtime on a webworker. So basically `run` runs on the main thread while `runAsync` can be a higher level interface that we use to run things on webworkers.

/cc @ntoll @antocuni  for your 2 cents :)

Reply:
This feels like we only needs a `run` method, with perhaps (at a later date) a flag like `webworker=true` to make it run "async". Like @fpliger I prefer a single Zen-o-Python one-and-obvious way to do this, and I think a future flag should be explicit about what it's providing (i.e. we call it `webworker` rather than `async` because the former is more descriptive, even though the latter is "true").

Reply:
The reason I was thinking about using `run` and `runAsync` (or maybe `runOnWorker` or something like that) is because of what you and @hoodmane mentioned below.... The 2 have **significant** differences and I was expecting for it to be too much to cramp everything in 1 holy method. 

That said, I think we should defer the decision until we actually have a plan on how we want to support messaging between the worker and the main thread, how we'll support writing to the dom, etc...

Reply:
decided on `runPythonAsync` 🗡️